hypervisor/src/cpu.rs

```diff
@@ -295,6 +322,37 @@ pub trait Vcpu: Send + Sync {
     #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
     fn get_one_reg(&self, reg_id: u64) -> Result<u64>;
     ///
+    /// Gets a list of the guest registers that are supported for the
+    /// KVM_GET_ONE_REG/KVM_SET_ONE_REG calls.
+    ///
+    #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+    fn get_reg_list(&self, reg_list: &mut RegList) -> Result<()>;
+    ///
+    /// Save the state of the core registers.
+    ///
+    #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+    fn save_core_registers(&self, state: &mut kvm_regs) -> Result<()>;
+    ///
+    /// Restore the state of the core registers.
+    ///
+    #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+    fn restore_core_registers(&self, state: &kvm_bindings::kvm_regs) -> Result<()>;
```

Comment:
This trait is not KVM specific so I would rather not see any KVM specific code here.

I think you can use `StandardRegisters` here.

Reply:
Thanks for pointing this out @liuw. Indeed I did not think of the Hyper-V stuff for the hypervisor crate. I've changed the `kvm_regs` to `StandardRegisters`. Hope this would address your concerns.